code/modules/clothing/spacesuits/rig/modules/specific/ai_container.dm

```diff
@@ -213,3 +228,237 @@
+			H.setToxLoss(0)
+
+			if(paddles && !paddles.busy)
+				if(H.isSynthetic())
+					paddles.use_on_synthetic = TRUE
+				else
+					paddles.use_on_synthetic = FALSE
+
+				paddles.busy = TRUE
+				paddles.do_revive(H, integrated_ai)
+				paddles.busy = FALSE
+				if(H.stat != DEAD)
+					corpse_pilot = TRUE
+
+		if(H.stat != DEAD)
+			if(corpse_pilot)
+				H.setOxyLoss(0)
+				H.add_modifier(/datum/modifier/risen_corpse, 30 SECONDS, suppress_failure = TRUE)
+				last_defib = world.time
+			if(H.sleeping)
+				H.AdjustSleeping(-10)
+
+		if(H.has_modifier_of_type(/datum/modifier/risen_corpse))
+			H.setToxLoss(min(H.getToxLoss(), 30))
+			H.setOxyLoss(min(H.getOxyLoss(), 20))
+
+	. = ..()
+
+/obj/item/rig_module/ai_container/advanced/proc/jack_brain()
+	set category = "Hardsuit"
+	set name = "Enable Direct Control"
+	set desc = "Engage a neural jack."
+
+	var/mob/living/jacker = integrated_ai
+	var/obj/item/weapon/rig/rig = holder
+	var/mob/living/carbon/human/H = rig.wearer
+	if(!(istype(H) && H.get_rig() == rig))
+		to_chat(jacker, "Your rig does not have a pilot, or is attached to an incompatible bioform.")
+		return
+
+	if(jacker.stat)
+		to_chat(src, "You cannot do that in your current state.")
+		return
+
+	if(!holder.ai_override_enabled)
+		to_chat(src, "You are locked out of \the [src]'s neural jack controls.")
+		return
+
+	to_chat(jacker, "You begin the process of enabling \the [src]'s neural jack.")
+	to_chat(H, "<span class='warning'>\The [rig]'s [src] begins clacking, accompanied by a pressure along your spine.</span>")
+
+	if(do_after(jacker, 1 MINUTE, H, ignore_movement = TRUE))
+		if(!H || !integrated_ai || controlling || !rig.ai_override_enabled)
+			return
+		else
+
+			to_chat(jacker, "<span class='warning'>You fully engage \the [src]'s neural jack, interfacing directly with the pilot's nervous system.</span>")
+			var/obj/item/organ/internal/brain/Brain = H.internal_organs_by_name[O_BRAIN]
+			if(!Brain)
+				to_chat(jacker,"\The [src] responds with an error code, as the neural jack swiftly retracts. The pilot has no neural cortex.")
+				return
+
+			if(H.has_brain_worms())
+				to_chat(jacker, "<span class='warning'>\The [src] connects, but only for a moment. You see the world through.. alien eyes, for a brief instant, as the jack retracts.</span>")
+				to_chat(H, "<span class='warning'>Something moves under the surface of your neck.</span>")
+				return
+
+			if(Brain.robotic < ORGAN_ASSISTED)
+				to_chat(H, "<span class='warning'>You feel a strange shifting sensation behind your eyes as a consciousness displaces yours. Your final sensation is that of something wet tearing.</span>")
+				H.adjustBrainLoss(rand(0,5))
+				imperfect_connection = TRUE
+
+			else
+				to_chat(H, "<span class='warning'>You feel a strange shifting sensation behind your eyes as a consciousness displaces yours.</span>")
+
+			if(H.stat == DEAD)
+				corpse_pilot = TRUE
+
+			// pilot -> brain
+			var/h2b_id = H.computer_id
```

Comment:
This code and the brain transfer code in general should be generalized into a proc, maybe on the captive_brain, rather than copypasted.